servers/rendering/renderer_geometry_instance.h

```diff
@@ -122,6 +123,9 @@ class RenderGeometryInstanceBase : public RenderGeometryInstance {
 		bool cast_double_sided_shadows = false;
 		bool dirty_dependencies = false;
 
+		float lightmap_scale = 1.0;
+		int32_t lightmap_size_global_uniform_pos = -2; //signifies not allocated, -1 is used for tried to allocate but failed
```

Comment:
This can default to -1 now and the comment can be removed

Reply:
I think we still need this; otherwise, we'll keep trying to allocate even when we get a `-1` from the allocation function, resulting in a spammed error.

Reply:
Right, in that case we need to change this a little bit ``global_shader_uniforms_unit_variable_update()`` should error if a value less than one is passed into it instead of silently returning. 

Also when setting ``instance_data.instance_uniforms_ofs`` we need to make sure we aren't passing a negative index

I'll leave a few more comments